### ISSUE TITLE
duplicates of jupyter-server-proxy's launcher icons

### ISSUE THREAD

**Issue by jhgoebbert** [cmd-ntrf/jupyter-lmod#30]
Hello,  

we see duplicates of jupyter-server-proxy's launcher icons in the JupyterLab panel.  

It is caused by jupyter-lmod and its function [setup_proxy_commands](https://github.com/cmd-ntrf/jupyter-lmod/blob/2.0.1/jupyterlab/src/index.ts#L236)()
This function seems to be a duplicate of [this code](https://github.com/jupyterhub/jupyter-server-proxy/blob/master/jupyterlab-server-proxy/src/index.ts#L78) of the jupyter-server-proxy extension.

Any idea how to avoid this dublicates of icons in the launcher panel?

**Comment by jhgoebbert** [comment]
![jupyterlmod-bug](https://user-images.githubusercontent.com/9266792/99593844-9bf0b500-29f2-11eb-9e5a-1c23cef46c53.png)

**Comment by jhgoebbert** [comment]
The left "Xpra Desktop" icons comes from `jupyter-server-proxy` and the right "Xpra Desktop" icon is created by `jupyter-lmod` from the same json data.

**Comment by cmd-ntrf** [comment]
You can disable the icon added by `jupyter-server-proxy` by defining `"enabled": False` in the Xpra Desktop launcher entry setting. [Point 7 of jupyter-server-proxy server process options documentation](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html#server-process-options)

**Comment by jhgoebbert** [comment]
Hi cmd-ntrf,

thanks for the quick answer, but unfortunately it doesn't solve the problem as I don't see any of the two icons anymore with the suggested solution.  

Currently `jupyter-server-proxy` and `jupyter-lmod` interprets the same json-Data at `https://.../server-proxy/servers-info`, which results in two icons for the same server-proxy - in my case it is my xprahtml5-proxy. If I would set "enable": False" then `jupyter-server-proxy` and `jupyter-lmod` would both hide the icon and I have no icon at all.

The question is, how to ensure, that the icon appears only once.

**Comment by cmd-ntrf** [comment]
jupyter-lmod add icons to JupyterLab launcher only if there is a module with the same name as the jupyter-server-proxy entry. This is done in the [`update_launcher`](https://github.com/cmd-ntrf/jupyter-lmod/blob/master/jupyterlab/src/index.ts#L39) function. jupyter-lmod does not consider the value of the `enabled` parameter.

The `setup_proxy_commands` function do not append anything to the launcher. It only build a data structure (`server_proxy_infos`) containing launcher items that will be added to the launcher in the event of a module load.

Therefore, if by setting `enabled` to False, both icons dissapear, one icon came from jupyter-server-proxy and the second icon came from something else, most likely your own xprahtml5-proxy plugin.

**Comment by jhgoebbert** [comment]
Ok, I see. `jupyter-lmod` replaces the functionality of `jupyter-server-proxy` and does the job of adding the launcher icons. Thank you for this clarification.

Currently I need to patch `jupyter-server-proxy` to allow passing url-parameters (https://github.com/jupyterhub/jupyter-server-proxy/pull/226) to the link of the launcher icon: [`const url = PageConfig.getBaseUrl() + server_process.name + '/' + urlfile;`](https://github.com/jupyterhub/jupyter-server-proxy/pull/226/files#diff-0a45a00aa865ac00aee8ce0fc05e45551ff22eadda3b623db7bf7e313a8e2cbbR86)

Then I need to patch `jupyter-lmod` the same way to consider the  `urlfile`, I suppose.

**Comment by cmd-ntrf** [comment]
Fixed in release 3.0.0
